### PR TITLE
testing zk authentication

### PR DESCRIPTION
**Pull request by anshupande**

### CODE REVIEW

File: v2/fleet-local/control/zk-exhibitor@.service [hunk 1]
```diff
@@ -25,6 +25,7 @@ ExecStart=/usr/bin/sh -c "sudo /usr/bin/docker run \
   -e S3_BUCKET=`etcdctl get /zookeeper/s3_exhibitor_bucket` \
   -e S3_PREFIX=`etcdctl get /zookeeper/s3_exhibitor_prefix` \
   -e ZK_ENSEMBLE_SIZE=`etcdctl get /zookeeper/ensemble_size` \
+  -e ZK_PASSWORD=`etcdctl get /zookeeper/ZK_PASSWORD` \
```
Comment (iderdik) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: What is the username?

Comment (anshupande) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: Let me add to secrets

Comment (anshupande) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: Should I we add new secrets under Ethos-adobe-platform in secrets or behance
?

Comment (anshupande) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: https://secrets.cloudops.adobe.com/SecretView.aspx?secretid=21521

**Comment by mannytoledo** [comment]
If it works then it looks good to me

**Comment by anshupande** [comment]
screen shot 2016-03-01 at 3 53 51 pm

File: v2/fleet-local/control/zk-exhibitor@.service (hunk 1 above)
Comment (iderdik) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: What happens if the password is blank or not set?

Comment (anshupande) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: I tried setting it to blank via jenkins:

`etcdctl get zookeeper/ZK_PASSWORD == ****
etcdctl set zookeeper/ZK_PASSWORD ''
Error:  value required`

Comment (anshupande) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: I will test it for the scenario where it is not set, for which I believe startup will fail as we have this line in startup: `-e ZK_PASSWORD=`etcdctl get /zookeeper/ZK_PASSWORD` `

Comment (iderdik) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: The Jenkins job doesn't let you set blanks (cc/ @chr0n1x ). However, you can test by setting it to blank via the `etcdctl` cli tool on a control node

Comment (anshupande) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: It does not allow to set it to blank even from CLI:

anshu@ip-10-43-1-211 sandbox-anshu-stack-us-east-1a-control ~ $ etcdctl set zookeeper/ZK_PASSWORD

Comment (anshupande) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: I can do a A/B and test both scenarios

Comment (iderdik) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: Try `etcdctl set zookeeper/ZK_PASSWORD ''`

From: Anshu Pande
Reply-To: adobe-platform/mesos-systemd
Date: Wednesday, March 2, 2016 at 9:35 AM
To: adobe-platform/mesos-systemd
Cc: Israel Derdik
Subject: Re: [mesos-systemd] testing zk authentication (#146)

In v2/fleet-local/control/zk-exhibitor@.servicehttps://github.com/adobe-platform/mesos-systemd/pull/146#discussion_r54729893:

> @@ -25,6 +25,7 @@ ExecStart=/usr/bin/sh -c "sudo /usr/bin/docker run \
>    -e S3_BUCKET=`etcdctl get /zookeeper/s3_exhibitor_bucket` \
>    -e S3_PREFIX=`etcdctl get /zookeeper/s3_exhibitor_prefix` \
>    -e ZK_ENSEMBLE_SIZE=`etcdctl get /zookeeper/ensemble_size` \
> -  -e ZK_PASSWORD=`etcdctl get /zookeeper/ZK_PASSWORD` \

It does not allow to set it to blank even from CLI:

anshu@ip-10-43-1-211 sandbox-anshu-stack-us-east-1a-control ~ $ etcdctl set zookeeper/ZK_PASSWORD

Comment (anshupande) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: So I tested with blank password and it does not show the login page and just displays the UI

Comment (anshupande) [inline v2/fleet-local/control/zk-exhibitor@.service:28]: And lastly I did `core@ip-10-43-4-125 sandbox-anshu-stack-us-east-1e-control ~ $ etcdctl rm zookeeper/ZK_PASSWORD
PrevNode.Value: 
core@ip-10-43-4-125 sandbox-anshu-stack-us-east-1e-control ~ $ etcdctl get zookeeper/ZK_PASSWORD
Error:  100: Key not found (/zookeeper/ZK_PASSWORD) [208101]`

I was able to access the console without being prompted for username and password

**Comment by iderdik** [comment]
:+1:

**Comment by anshupande** [comment]
I will add secrets ( zk-exhibitor) for each account and then merge it

**Comment by chr0n1x** [comment]
mesos & marathon are still up? one can hit the other?

**Comment by anshupande** [comment]
@chr0n1x They are still up `3f42fd67c6f9        mesosphere/marathon:v0.11.1                                                                  "./bin/start --master"   10 minutes ago      Up 10 minutes                                                                                                        marathon
1d132edb8d48        mesosphere/mesos-master:0.25.0-0.2.70.ubuntu1404                                             "mesos-master --clust"   10 minutes ago      Up 10 minutes                                                                                                        mesos_master`

**Comment by anshupande** [comment]
@chr0n1x What is the best way to check if they can hit each other?

**Comment by iderdik** [comment]
See if the slaves register in the Mesos UI. Try deploying something.

From: Anshu Pande
Reply-To: adobe-platform/mesos-systemd
Date: Wednesday, March 2, 2016 at 12:14 PM
To: adobe-platform/mesos-systemd
Cc: Israel Derdik
Subject: Re: [mesos-systemd] testing zk authentication (#146)

@chr0n1xhttps://github.com/chr0n1x What is the best way to check if they can hit each other?

**Comment by anshupande** [comment]
Strangely when I am removing zk://username:password@host1:port1,host2:port2,.../path from --master in slave config the slave startup which was flapping when I used username:password

**Comment by anshupande** [comment]
screen shot 2016-03-07 at 6 01 54 pm

**Comment by anshupande** [comment]
My Bad...My fork was messed up...it had some bad escape sequences... I created a new clone and slave started with zoo auth https://github.com/adobe-platform/mesos-systemd/pull/150

**Comment by anshupande** [comment]
merged https://github.com/adobe-platform/mesos-systemd/pull/150 so safe to delete this duplicate

**Pull request closed without merging**